core/libraries/form_sections/form_handlers/FormHandler.php

```diff
@@ -587,7 +587,7 @@ public function display()
         ) {
             $form_html .= $this->form()->form_open($this->formAction());
         }
-        $form_html .= $this->form(true)->get_html($this->form_has_errors);
+        $form_html .= $this->form(true)->get_html();
```

Comment:
why are you making this change?
this could totally affect how other forms function?

Reply:
I tried to explain that in the commit message (https://github.com/eventespresso/event-espresso-core/pull/470/commits/f86313d72b5fc5a8a3134b9543e3b5ac1640bc27).
But basically, it seemed to me this logic was wrong: if the form was submitted on a previous request and found to be invalid, and then the invalid data was put into the session, then on the subsequent request, when this line gets ran as we're fetching the HTML to display the form, the instantiation of the form **does not yet have any errors**. 
And this logic is saying "well, if there are no errors in the form at the moment, ignore the session data, just display the form normally." So, even though there is stored invalid form data in the session, this code is saying to NOT check the session because we are unaware of any errors in the form.

Of course, if we were to only look in the session data, we'd see there is invalid form data in there. 

So I believe this new logic is right because just because the form hasn't been marked as "invalid" (ie, has validation errors) during THIS request, doesn't mean it wasn't marked as invalid during a previous request and that invalid data was stuffed into the session.

So this new logic basically says "regardless of whether we've found the form to be valid or invalid so far during this request, we still need to check in the session for invalid form data, because if there is we should display it."

Does that make sense?

And from what I understand, yes this will affect other forms, because the previous logic seemed like it was buggy (but you would know if somehow other code was relying on this logic.)

Reply:
> I tried to explain that in the commit message

hmmmm... ya that's stinky... GitHub doesn't always show all of the individual commit notes when reviewing because it kinda amalgamates all of the changes into one big final diff.

> Does that make sense?

roger Roger